projects/portal/src/app/models/cosmos/staking.service.ts

```diff
@@ -182,6 +182,148 @@ export class StakingService {
+    minimumGasPrice: proto.cosmos.base.v1beta1.ICoin,
+    privateKey: Uint8Array,
+    gasRatio: number,
+  ): Promise<SimulatedTxResultResponse> {
+    const dummyFee: proto.cosmos.base.v1beta1.ICoin = {
+      denom: minimumGasPrice.denom,
+      amount: '1',
+    };
+    const dummyGas: proto.cosmos.base.v1beta1.ICoin = {
+      denom: minimumGasPrice.denom,
+      amount: '1',
+    };
+    console.log('debug_in sim1', dummyGas);
+    const simulatedTxBuilder = await this.buildEditValidator(
+      keyType,
+      editValidatorData,
+      dummyGas,
+      dummyFee,
+      privateKey,
+    );
+    console.log('debug_in sim2_before');
+    const A = await this.txCommonService.simulateTx(simulatedTxBuilder, minimumGasPrice, gasRatio);
+    console.log('debug_in sim2_error_above_code', A);
+    return A;
+  }
+
+  async buildEditValidator(
+    keyType: KeyType,
+    editValidatorData: EditValidatorData,
+    gas: proto.cosmos.base.v1beta1.ICoin,
+    fee: proto.cosmos.base.v1beta1.ICoin,
+    privateKey: Uint8Array,
+  ): Promise<cosmosclient.TxBuilder> {
+    const sdk = await this.cosmosSDK.sdk().then((sdk) => sdk.rest);
+    const privKey = createCosmosPrivateKeyFromUint8Array(keyType, privateKey);
+    if (!privKey) {
+      throw Error('Invalid privateKey!');
+    }
+    const pubKey = privKey.pubKey();
+    const accAddress = cosmosclient.AccAddress.fromPublicKey(pubKey);
+    const valAddress = cosmosclient.ValAddress.fromPublicKey(pubKey);
+
+    // get account info
+    const account = await rest.auth
+      .account(sdk, accAddress)
+      .then((res) => res.data.account && cosmosclient.codec.unpackCosmosAny(res.data.account))
+      .catch((_) => undefined);
+
+    const baseAccount = convertUnknownAccountToBaseAccount(account);
+
+    if (!baseAccount) {
+      throw Error('Unused Account or Unsupported Account Type!');
+    }
+
+    if (editValidatorData.delegator_address !== accAddress.toString()) {
+      throw Error('delegator_address mismatch!');
+    }
+
+    if (editValidatorData.validator_address !== valAddress.toString()) {
+      throw Error('validator_address mismatch!');
+    }
+    console.log('debug_1-1');
+    // build tx ... Note: commission percent rate values are converted here.
+    const editValidatorTxData = {
+      description: {
+        moniker: editValidatorData.moniker,
+        identity: editValidatorData.identity,
+        website: editValidatorData.website,
+        security_contact: editValidatorData.security_contact,
+        details: editValidatorData.details,
+      },
+      validator_address: editValidatorData.validator_address,
+      /*
+      commission: {
+        rate: `${editValidatorData.rate}${'0000000000000000'}`,
+      },
+      */
+      commission_rate: '0.10',
+      min_self_delegation: editValidatorData.min_self_delegation,
+      /*
+      value: {
+        denom: editValidatorData.denom,
+        amount: editValidatorData.amount,
+      },
+      */
+    };
```

Comment:
Is this "editValidatorTxData" OK?
it comes from "proto.d.ts"
( commisson_rate is on debug )
![image](https://user-images.githubusercontent.com/75844498/169309567-2679d1c1-9da9-48ac-9acf-0b4c2022f719.png)

Reply:
1. I guess you should refer cosmos.staking.v1beta1.IMsgEditValidator.
2. I guess you need to set commission_rate query parameter to be 0.10 like value and you need to convert it to "100000000000000000". this is strange but it seems 10% => 0.1 => 0.1 * 10^18 maybe that seems 18digits fixed style string.